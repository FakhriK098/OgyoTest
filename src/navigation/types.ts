import { NavigatorScreenParams } from '@react-navigation/native';

export type RootStackParamList = {
  MainTabs: NavigatorScreenParams<MainTabParamList>;
  Detail: { id: string; title: string; data?: any };
};

export type HomeStackParamList = {
  Detail: { id: string };
};

export type MainTabParamList = {
  Home: undefined;
  User: undefined;
};
