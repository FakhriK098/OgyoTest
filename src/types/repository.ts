export interface IOwner {
  login: string;
  id: number;
  node_id: string;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: string;
  user_view_type: string;
  site_admin: boolean;
}

export interface IRepository {
  id: number;
  node_id: string;
  name: string;
  full_name: string;
  private: boolean;
  owner: IOwner;
  html_url: string;
  description: string;
}

export interface IFetchRepositoriesPayload {
  since?: number;
}

export interface ISearchRepositoriesPayload {
  query: string;
}

export interface ILinkHeader {
  next?: string;
  prev?: string;
  first?: string;
  last?: string;
}
