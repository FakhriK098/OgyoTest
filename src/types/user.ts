export interface IUser {
  id: number;
  node_id: string;
  name: string;
  full_name: string;
  private: boolean;
  owner: IUserOwner;
  html_url: string;
  description: string;
  size: number;
  stargazers_count: number;
  watchers_count: number;
  language: string;
  forks_count: number;
  open_issues_count: number;
  license: ILicense;
  topics: string[];
  network_count: number;
  subscribers_count: number;
}

export interface ILicense {
  key: string;
  name: string;
  spdx_id: string;
  url: string;
  node_id: string;
}

export interface IUserOwner {
  login: string;
  id: number;
  node_id: string;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: string;
  user_view_type: string;
  site_admin: boolean;
}

export interface IFetchUserPayload {
  fullName?: string;
}

export interface ILabelProps {
  title: string;
  value?: string;
}
