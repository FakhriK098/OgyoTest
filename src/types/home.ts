import { IRepository } from './repository';

export interface IFloatingActionButtonProps {
  onPress: () => void;
}

export interface ISearchBarProps {
  onSearch: (query: string) => void;
  onSelectSuggestion: (repository: IRepository) => void;
  onClear: () => void;
  suggestions: IRepository[];
  loading: boolean;
  placeholder?: string;
  onSubmitSearch: (query: string) => void;
}

export type TSortOrder = 'ASC' | 'DESC';

export interface ISortFilterModalProps {
  visible: boolean;
  onClose: () => void;
  onSort: (order: TSortOrder) => void;
  currentOrder: TSortOrder;
}
