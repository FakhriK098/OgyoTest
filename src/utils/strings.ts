import { ILinkHeader } from '../types/repository';

export const getFullName = (name: string): string => {
  return name.split('/').join(' ');
};

export const parseLinkHeader = (linkHeader?: string): ILinkHeader => {
  const links: ILinkHeader = {};
  if (!linkHeader) return links;

  const parts = linkHeader.split(',');
  parts.forEach(part => {
    const match = part.match(/<([^>]+)>;\s*rel="([^"]+)"/);
    if (match) {
      const [, url, rel] = match;
      links[rel as keyof ILinkHeader] = url;
    }
  });

  return links;
};

export const extractSinceFromUrl = (url?: string): number | null => {
  if (!url) return null;
  const match = url.match(/since=(\d+)/);
  return match ? parseInt(match[1], 10) : null;
};
