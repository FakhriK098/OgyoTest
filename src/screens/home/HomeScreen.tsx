import React, { useEffect, useState, useMemo } from 'react';
import {
  View,
  Text,
  StyleSheet,
  SafeAreaView,
  FlatList,
  ActivityIndicator,
  RefreshControl,
} from 'react-native';
import { colors } from '../../themes/colors';
import { useAppDispatch, useAppSelector } from '../../hooks/redux';
import {
  fetchRepositoriesRequest,
  fetchMoreRepositoriesRequest,
  fetchSearchRepositoriesRequest,
  clearSearchResults,
} from '../../store/slices/repositorySagaSlice';
import { IRepository } from '../../types/repository';
import CardRepository from './components/CardRepository';
import ItemSeparator from '../../components/ItemSeparator';
import FloatingActionButton from './components/FloatingActionButton';
import SearchBar from './components/SearchBar';
import { TSortOrder } from '../../types/home';
import SortFilterModal from './components/SortFilterModal';
import { ASC } from '../../utils/constants';

const HomeScreen = () => {
  const dispatch = useAppDispatch();
  const {
    repositories,
    searchRepositories,
    loading,
    loadingMore,
    searchLoading,
    error,
    hasMore,
  } = useAppSelector(state => state.repository);

  const [isSearching, setIsSearching] = useState(false);
  const [showSearchResults, setShowSearchResults] = useState(false);
  const [searchQuery, setSearchQuery] = useState('');
  const [sortOrder, setSortOrder] = useState<TSortOrder>(ASC);
  const [modalVisible, setModalVisible] = useState(false);

  useEffect(() => {
    dispatch(fetchRepositoriesRequest({ since: 0 }));
  }, [dispatch]);

  const sortedData = useMemo(() => {
    const dataToSort = showSearchResults ? searchRepositories : repositories;
    const sorted = [...dataToSort].sort((a, b) => {
      if (sortOrder === ASC) {
        return a.name.localeCompare(b.name);
      } else {
        return b.name.localeCompare(a.name);
      }
    });
    return sorted;
  }, [showSearchResults, searchRepositories, repositories, sortOrder]);

  const handleRefresh = () => {
    if (showSearchResults) {
      handleSubmitSearch(searchQuery);
    } else {
      dispatch(fetchRepositoriesRequest({ since: 0 }));
    }
  };

  const handleLoadMore = () => {
    if (!showSearchResults && !loadingMore && hasMore && !loading) {
      dispatch(fetchMoreRepositoriesRequest());
    }
  };

  const handleSearch = (query: string) => {
    setSearchQuery(query);
    if (query.trim().length > 0) {
      setIsSearching(true);
      dispatch(fetchSearchRepositoriesRequest({ query }));
    }
  };
  const handleSubmitSearch = (query: string) => {
    setSearchQuery(query);
    setShowSearchResults(true);
    setIsSearching(true);
    if (query.trim().length > 0) {
      dispatch(fetchSearchRepositoriesRequest({ query }));
    }
  };

  const handleSelectSuggestion = (repository: IRepository) => {
    setSearchQuery(repository.name);
    setShowSearchResults(true);
    setIsSearching(true);
    dispatch(fetchSearchRepositoriesRequest({ query: repository.name }));
  };

  const handleClearSearch = () => {
    setIsSearching(false);
    setShowSearchResults(false);
    setSearchQuery('');
    dispatch(clearSearchResults());
  };

  const handleOpenModal = () => {
    setModalVisible(true);
  };

  const handleCloseModal = () => {
    setModalVisible(false);
  };

  const handleSort = (order: TSortOrder) => {
    setSortOrder(order);
  };

  const renderItem = ({ item }: { item: IRepository }) => (
    <CardRepository item={item} />
  );

  const renderFooter = () => {
    if (!loadingMore) return null;
    return (
      <View style={styles.footerLoader}>
        <ActivityIndicator size="small" color={colors.shade800} />
      </View>
    );
  };

  const renderEmpty = () => {
    if (loading && !showSearchResults) {
      return (
        <View style={styles.emptyContainer}>
          <ActivityIndicator size="large" color={colors.shade800} />
          <Text style={styles.loadingText}>Loading repositories...</Text>
        </View>
      );
    }

    if (showSearchResults && searchLoading) {
      return (
        <View style={styles.emptyContainer}>
          <ActivityIndicator size="large" color={colors.shade800} />
          <Text style={styles.loadingText}>Searching...</Text>
        </View>
      );
    }

    if (
      showSearchResults &&
      searchRepositories.length === 0 &&
      !searchLoading
    ) {
      return (
        <View style={styles.emptyContainer}>
          <Text style={styles.noDataText}>No Data Found</Text>
        </View>
      );
    }

    if (error) {
      return (
        <View style={styles.emptyContainer}>
          <Text style={styles.errorText}>{error}</Text>
        </View>
      );
    }

    return null;
  };

  return (
    <SafeAreaView style={styles.container}>
      <SearchBar
        onSearch={handleSearch}
        onSelectSuggestion={handleSelectSuggestion}
        onClear={handleClearSearch}
        onSubmitSearch={handleSubmitSearch}
        suggestions={
          isSearching && !showSearchResults ? searchRepositories : []
        }
        loading={searchLoading}
        placeholder="Search GitHub repository..."
      />

      <FlatList
        data={sortedData}
        renderItem={renderItem}
        keyExtractor={item => item.id.toString()}
        ItemSeparatorComponent={ItemSeparator}
        showsVerticalScrollIndicator={false}
        style={styles.flatList}
        refreshControl={
          <RefreshControl
            refreshing={
              (loading && repositories.length === 0) ||
              (searchLoading && showSearchResults)
            }
            onRefresh={handleRefresh}
          />
        }
        onEndReached={handleLoadMore}
        onEndReachedThreshold={0.5}
        ListFooterComponent={renderFooter}
        ListEmptyComponent={renderEmpty}
      />

      <FloatingActionButton onPress={handleOpenModal} />

      <SortFilterModal
        visible={modalVisible}
        onClose={handleCloseModal}
        onSort={handleSort}
        currentOrder={sortOrder}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.brokenWhite,
  },
  header: {
    backgroundColor: colors.white,
    padding: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  flatList: {
    paddingVertical: 8,
    paddingHorizontal: 8,
  },
  footerLoader: {
    paddingVertical: 20,
    alignItems: 'center',
  },
  emptyContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 50,
  },
  errorText: {
    fontSize: 16,
    color: colors.shade800,
    textAlign: 'center',
  },
  loadingText: {
    marginTop: 10,
    fontSize: 14,
    color: colors.shade75,
  },
  noDataText: {
    fontSize: 20,
    fontWeight: '600',
    color: colors.shade800,
    marginBottom: 8,
  },
  noDataSubtext: {
    fontSize: 14,
    color: colors.shade75,
    textAlign: 'center',
  },
});

export default HomeScreen;
