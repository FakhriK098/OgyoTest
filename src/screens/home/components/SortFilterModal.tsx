import React from 'react';
import {
  Modal,
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  TouchableWithoutFeedback,
} from 'react-native';
import { colors } from '../../../themes/colors';
import { ISortFilterModalProps, TSortOrder } from '../../../types/home';
import { ASC, DESC } from '../../../utils/constants';

const SortFilterModal: React.FC<ISortFilterModalProps> = ({
  visible,
  onClose,
  onSort,
  currentOrder,
}) => {
  const handleSort = (order: TSortOrder) => {
    onSort(order);
    onClose();
  };

  return (
    <Modal
      animationType="fade"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <TouchableWithoutFeedback onPress={onClose}>
        <View style={styles.modalOverlay}>
          <TouchableWithoutFeedback>
            <View style={styles.modalContent}>
              <Text style={styles.modalTitle}>Sort Repositories</Text>

              <TouchableOpacity
                style={[
                  styles.option,
                  currentOrder === ASC && styles.selectedOption,
                ]}
                onPress={() => handleSort(ASC)}
              >
                <Text
                  style={[
                    styles.optionText,
                    currentOrder === ASC && styles.selectedText,
                  ]}
                >
                  Ascending (A-Z)
                </Text>
              </TouchableOpacity>

              <TouchableOpacity
                style={[
                  styles.option,
                  currentOrder === DESC && styles.selectedOption,
                ]}
                onPress={() => handleSort(DESC)}
              >
                <Text
                  style={[
                    styles.optionText,
                    currentOrder === DESC && styles.selectedText,
                  ]}
                >
                  Descending (Z-A)
                </Text>
              </TouchableOpacity>

              <TouchableOpacity style={styles.closeButton} onPress={onClose}>
                <Text style={styles.closeButtonText}>Close</Text>
              </TouchableOpacity>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContent: {
    backgroundColor: colors.white,
    borderRadius: 12,
    padding: 20,
    width: '80%',
    maxWidth: 320,
    elevation: 5,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
  },
  modalTitle: {
    fontSize: 20,
    fontWeight: '600',
    color: colors.black,
    marginBottom: 20,
    textAlign: 'center',
  },
  option: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderRadius: 8,
    marginBottom: 8,
    backgroundColor: colors.brokenWhite,
  },
  selectedOption: {
    backgroundColor: colors.shade75,
    borderWidth: 1,
    borderColor: colors.shade800,
  },
  optionText: {
    fontSize: 16,
    color: colors.shade800,
  },
  selectedText: {
    fontWeight: '600',
    color: colors.black,
  },
  closeButton: {
    marginTop: 12,
    paddingVertical: 12,
    backgroundColor: colors.shade800,
    borderRadius: 8,
    alignItems: 'center',
  },
  closeButtonText: {
    fontSize: 16,
    color: colors.white,
    fontWeight: '600',
  },
});

export default SortFilterModal;
