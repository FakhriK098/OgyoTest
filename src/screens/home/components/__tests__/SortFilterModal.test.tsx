import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SortFilterModal from '../SortFilterModal';
import { ISortFilterModalProps, TSortOrder } from 'src/types/home';
import { ASC, DESC } from 'src/utils/constants';

describe('SortFilterModal', () => {
  const defaultProps: ISortFilterModalProps = {
    visible: false,
    onClose: jest.fn(),
    onSort: jest.fn(),
    currentOrder: ASC,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly when visible', () => {
    const props = { ...defaultProps, visible: true };
    const { getByText } = render(<SortFilterModal {...props} />);

    expect(getByText('Sort Repositories')).toBeTruthy();
    expect(getByText('Ascending (A-Z)')).toBeTruthy();
    expect(getByText('Descending (Z-A)')).toBeTruthy();
    expect(getByText('Close')).toBeTruthy();
  });

  it('should not render when not visible', () => {
    const props = { ...defaultProps, visible: false };
    const { queryByText } = render(<SortFilterModal {...props} />);
    expect(queryByText('Sort Repositories')).toBeNull();
  });

  it('should highlight ascending option when currentOrder is ASC', () => {
    const props = { ...defaultProps, visible: true, currentOrder: ASC };
    const { getByText } = render(<SortFilterModal {...props} />);

    const ascendingOption = getByText('Ascending (A-Z)');
    expect(ascendingOption).toBeTruthy();
  });

  it('should highlight descending option when currentOrder is DESC', () => {
    const props = { ...defaultProps, visible: true, currentOrder: DESC };
    const { getByText } = render(<SortFilterModal {...props} />);

    const descendingOption = getByText('Descending (Z-A)');
    expect(descendingOption).toBeTruthy();
  });

  it('should call onSort with ASC and close modal when ascending is selected', () => {
    const onSortMock = jest.fn();
    const onCloseMock = jest.fn();
    const props = {
      ...defaultProps,
      visible: true,
      currentOrder: DESC,
      onSort: onSortMock,
      onClose: onCloseMock,
    };

    const { getByText } = render(<SortFilterModal {...props} />);

    const ascendingOption = getByText('Ascending (A-Z)');
    fireEvent.press(ascendingOption);

    expect(onSortMock).toHaveBeenCalledWith(ASC);
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('should call onSort with DESC and close modal when descending is selected', () => {
    const onSortMock = jest.fn();
    const onCloseMock = jest.fn();
    const props = {
      ...defaultProps,
      visible: true,
      currentOrder: ASC,
      onSort: onSortMock,
      onClose: onCloseMock,
    };

    const { getByText } = render(<SortFilterModal {...props} />);

    const descendingOption = getByText('Descending (Z-A)');
    fireEvent.press(descendingOption);

    expect(onSortMock).toHaveBeenCalledWith(DESC);
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('should call onClose when close button is pressed', () => {
    const onCloseMock = jest.fn();
    const props = { ...defaultProps, visible: true, onClose: onCloseMock };

    const { getByText } = render(<SortFilterModal {...props} />);

    const closeButton = getByText('Close');
    fireEvent.press(closeButton);

    expect(onCloseMock).toHaveBeenCalled();
  });

  it('should call onClose when modal backdrop is pressed', () => {
    const onCloseMock = jest.fn();
    const props = { ...defaultProps, visible: true, onClose: onCloseMock };

    const { UNSAFE_getAllByType } = render(<SortFilterModal {...props} />);

    const TouchableWithoutFeedback =
      require('react-native').TouchableWithoutFeedback;
    const touchables = UNSAFE_getAllByType(TouchableWithoutFeedback);

    if (touchables.length > 0) {
      fireEvent.press(touchables[0]);
      expect(onCloseMock).toHaveBeenCalled();
    }
  });

  it('should handle onRequestClose callback', () => {
    const onCloseMock = jest.fn();
    const props = { ...defaultProps, visible: true, onClose: onCloseMock };

    const { UNSAFE_getByType } = render(<SortFilterModal {...props} />);

    const Modal = require('react-native').Modal;
    const modal = UNSAFE_getByType(Modal);

    if (modal.props.onRequestClose) {
      modal.props.onRequestClose();
      expect(onCloseMock).toHaveBeenCalled();
    }
  });

  it('should prevent event bubbling on modal content touch', () => {
    const onCloseMock = jest.fn();
    const props = { ...defaultProps, visible: true, onClose: onCloseMock };

    const { getByText } = render(<SortFilterModal {...props} />);

    const modalTitle = getByText('Sort Repositories');
    fireEvent.press(modalTitle);

    expect(onCloseMock).toHaveBeenCalled();
  });

  it('should handle multiple rapid presses on sort options', () => {
    const onSortMock = jest.fn();
    const onCloseMock = jest.fn();
    const props = {
      ...defaultProps,
      visible: true,
      onSort: onSortMock,
      onClose: onCloseMock,
    };

    const { getByText } = render(<SortFilterModal {...props} />);

    const ascendingOption = getByText('Ascending (A-Z)');

    fireEvent.press(ascendingOption);
    fireEvent.press(ascendingOption);
    fireEvent.press(ascendingOption);

    expect(onSortMock).toHaveBeenCalledTimes(3);
    expect(onCloseMock).toHaveBeenCalledTimes(3);
  });

  it('should handle switching between sort options', () => {
    const onSortMock = jest.fn();
    const onCloseMock = jest.fn();
    let props = {
      ...defaultProps,
      visible: true,
      currentOrder: ASC as TSortOrder,
      onSort: onSortMock,
      onClose: onCloseMock,
    };

    const { getByText, rerender } = render(<SortFilterModal {...props} />);

    const descendingOption = getByText('Descending (Z-A)');
    fireEvent.press(descendingOption);

    expect(onSortMock).toHaveBeenCalledWith(DESC);
    expect(onCloseMock).toHaveBeenCalled();

    jest.clearAllMocks();
    props = { ...props, currentOrder: DESC };
    rerender(<SortFilterModal {...props} />);

    const ascendingOption = getByText('Ascending (A-Z)');
    fireEvent.press(ascendingOption);

    expect(onSortMock).toHaveBeenCalledWith(ASC);
    expect(onCloseMock).toHaveBeenCalled();
  });

  it('should maintain modal visibility state correctly', () => {
    let props = { ...defaultProps, visible: false };
    const { queryByText, rerender } = render(<SortFilterModal {...props} />);

    expect(queryByText('Sort Repositories')).toBeNull();

    props = { ...props, visible: true };
    rerender(<SortFilterModal {...props} />);

    expect(queryByText('Sort Repositories')).toBeTruthy();

    props = { ...props, visible: false };
    rerender(<SortFilterModal {...props} />);

    expect(queryByText('Sort Repositories')).toBeNull();
  });

  it('should display correct modal animation type', () => {
    const props = { ...defaultProps, visible: true };
    const { UNSAFE_getByType } = render(<SortFilterModal {...props} />);

    const Modal = require('react-native').Modal;
    const modal = UNSAFE_getByType(Modal);

    expect(modal.props.animationType).toBe('fade');
    expect(modal.props.transparent).toBe(true);
  });

  it('should handle edge cases with undefined callbacks', () => {
    const consoleSpy = jest.spyOn(console, 'error').mockImplementation();

    const props = {
      ...defaultProps,
      visible: true,
      onSort: undefined as any,
      onClose: undefined as any,
    };

    const { getByText } = render(<SortFilterModal {...props} />);

    expect(() => {
      fireEvent.press(getByText('Ascending (A-Z)'));
    }).toThrow();

    expect(() => {
      fireEvent.press(getByText('Close'));
    }).not.toThrow();

    consoleSpy.mockRestore();
  });

  it('should handle all TSortOrder values', () => {
    const onSortMock = jest.fn();
    const props = {
      ...defaultProps,
      visible: true,
      onSort: onSortMock,
    };

    const { getByText } = render(<SortFilterModal {...props} />);

    fireEvent.press(getByText('Ascending (A-Z)'));
    expect(onSortMock).toHaveBeenCalledWith('ASC');

    jest.clearAllMocks();

    fireEvent.press(getByText('Descending (Z-A)'));
    expect(onSortMock).toHaveBeenCalledWith('DESC');
  });

  it('should match snapshot when not visible', () => {
    const props = { ...defaultProps, visible: false };
    const { toJSON } = render(<SortFilterModal {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when visible with ASC selected', () => {
    const props = {
      ...defaultProps,
      visible: true,
      currentOrder: ASC as TSortOrder,
    };
    const { toJSON } = render(<SortFilterModal {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when visible with DESC selected', () => {
    const props = {
      ...defaultProps,
      visible: true,
      currentOrder: DESC as TSortOrder,
    };
    const { toJSON } = render(<SortFilterModal {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
