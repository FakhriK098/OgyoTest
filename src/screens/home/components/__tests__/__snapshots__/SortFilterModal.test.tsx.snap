// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortFilterModal should match snapshot when not visible 1`] = `null`;

exports[`SortFilterModal should match snapshot when visible with ASC selected 1`] = `
<Modal
  animationType="fade"
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "elevation": 5,
          "maxWidth": 320,
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Sort Repositories
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#DEDEDE",
            "borderColor": "#161B28",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#161B28",
                "fontSize": 16,
              },
              {
                "color": "#000000",
                "fontWeight": "600",
              },
            ]
          }
        >
          Ascending (A-Z)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EEEBE3",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#161B28",
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Descending (Z-A)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#161B28",
            "borderRadius": 8,
            "marginTop": 12,
            "opacity": 1,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`SortFilterModal should match snapshot when visible with DESC selected 1`] = `
<Modal
  animationType="fade"
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "elevation": 5,
          "maxWidth": 320,
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Sort Repositories
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#EEEBE3",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#161B28",
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Ascending (A-Z)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#DEDEDE",
            "borderColor": "#161B28",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#161B28",
                "fontSize": 16,
              },
              {
                "color": "#000000",
                "fontWeight": "600",
              },
            ]
          }
        >
          Descending (Z-A)
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#161B28",
            "borderRadius": 8,
            "marginTop": 12,
            "opacity": 1,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
